bin/echo.c

```diff
@@ -40,16 +40,53 @@ void print_s2n_error(const char *app_error)
             s2n_strerror_debug(s2n_errno, "EN"));
 }
 
-int negotiate(struct s2n_connection *conn)
+/* Poll the given file descriptor for an event determined by the blocked status */
+static int wait_for_event(int fd, s2n_blocked_status blocked)
+{
+    struct pollfd reader = { .fd = fd, .events = 0 };
+
+    switch (blocked) {
+    case S2N_NOT_BLOCKED:
+        return S2N_SUCCESS;
+        break;
+    case S2N_BLOCKED_ON_READ:
+        reader.events |= POLLIN;
+        break;
+    case S2N_BLOCKED_ON_WRITE:
+        reader.events |= POLLOUT;
+        break;
+    case S2N_BLOCKED_ON_APPLICATION_INPUT:
+        /* This case is not encountered by the s2nc/s2nd applications,
+         * but is detected for completeness */
+        return S2N_SUCCESS;
+        break;
```

Comment:
nit: do you need a break if you're returning?

Reply:
It seems no, I thought this resulted in a compiler warning. Removed!